_tour/multiple-parameter-lists.md

```diff
@@ -0,0 +1,44 @@
+---
+layout: tour
+title: Multiple Parameter Lists
+
+discourse: true
+
+partof: scala-tour
+
+num: 10
+next-page: case-classes
+previous-page: nested-functions
+
+redirect_from: "/tutorials/tour/multiple-parameter-lists.html"
+---
+
+Methods may define multiple parameter lists. When a method is called with a fewer number of parameter lists, then this will yield a function taking the missing parameter lists as its arguments.
+
+Here is an example, defined in [Traversable](/overviews/collections/trait-traversable.html) trait from Scala collections:
+
+```
+def foldLeft[B](z: B)(op: (B, A) => B): B
+```
+
+`foldLeft` applies a binary operator `op` to an initial value `z` and all elements of this traversable, going left to right. Here is an example of its usage:
```

Comment:
I would mention here one of the primary reasons why currying is useful: to fix the parameter `z` and reuse it in the call site several times. For example, when we're using the same parameter `z` all the time, it's better to reuse the same function.

I'm trying to find a good example that illustrates this scenario, but cannot come up with a good one. Do you happen to have a good one in mind? :smile:

Reply:
I think that being able to reuse the parameter is just one of the advantages here.
Another is the special syntax you can get, `list.foldLeft(z) { (acc, value) => /**/ }`.

A bigger advantage is, though, that `scalac` doesn't use type information in one parameter for another parameter's inference.
With a `def foldLeft[B](z: B, op: (B, A) => B)`, you can't write `List(1, 2).foldLeft(0, _ + _)`. It complains about the first `_` having a missing parameter type.

Reply:
> A bigger advantage is, though, that scalac doesn't use type information in one parameter for another parameter's inference.